File: src/main/scala/org/muieer/scala/leetcode/LeetCode876.scala
package org.muieer.scala.leetcode

/*
* 876. 链表的中间结点 https://leetcode.cn/problems/middle-of-the-linked-list/
* */
object LeetCode876 {

  def middleNode(head: ListNode): ListNode = {

    if (head.next == null) return head

    var slow = head
    var fast = head

    while(fast != null && fast.next != null) {
      slow = slow.next
      fast = fast.next.next
    }

    slow
  }

}

File: src/main/java/org/muieer/java/leetcode/LeetCode876.java
package org.muieer.java.leetcode;

/*
 * 876. 链表的中间结点 https://leetcode.cn/problems/middle-of-the-linked-list/
 * */
public class LeetCode876 {

    public ListNode middleNode(ListNode head) {

        if (head.next == null) return head;

        ListNode slow = head;
        ListNode fast = head;

        while (fast != null && fast.next != null) {
            slow = slow.next;
            fast = fast.next.next;
        }

        return slow;
    }
}
